arch/arc/core/smp.c

```diff
@@ -131,17 +131,19 @@ static void sched_ipi_handler(const void *unused)
 }
 
 /* arch implementation of sched_ipi */
-void arch_sched_ipi(void)
+void arch_sched_ipi(uint32_t cpu_bitmap)
 {
 	uint32_t i;
 
-	/* broadcast sched_ipi request to other cores
+	/* Send sched_ipi request to other cores
 	 * if the target is current core, hardware will ignore it
 	 */
-	unsigned int num_cpus = arch_num_cpus();
+	uint32_t num_cpus = (uint32_t)arch_num_cpus();
```

Comment:
arch_num_cpus() is a specified API that returns an unsigned int.  Why muck with typing here?

Reply:
Just addressing a previous comment about mixing uint32_t and unsigned int types. I will look again at this. There is probably a cleaner way.